### PR TITLE
Used dict instead of typing.Dict in most of the files.

### PR DESCRIPTION
**Pull request by webbdays**
Fix for issue #2634 .

### CODE REVIEW

**Comment by marco-c** [comment]
Thanks, the change looks good. Could you also apply the same change for `list` and `tuple` in separate commits of this PR?

**Comment by webbdays** [comment]
Ok

**Comment by webbdays** [comment]
@marco-c 

Is there any way in python to perform in-place replace.
Suppose ,
File 1 has these lines
Line1a
Line2b
Line3a

I want to just change Line2b to Line2a in that file.

In python we read file to a string
Modify the string and then write to the file.
But for git diff it is shown as deleting whole file and writing to the file.
-Line1a
-Line2b
-Line3a
+Line1a
+Line2a
+Line3a

I want to just
-Line2b
+Line2a

Which is possible to do in nano text editor.

But I want to do the same for multiple files. Like write some python program or any other way.

Please help.

**Comment by webbdays** [comment]
i get to know this on stack overflow,
which performs replace recursively in dir and it's sub dirs.
`find . -type f -readable -writable -exec sed   
-i "s/string/nstring/g" {} \;`

**Comment by webbdays** [comment]
Ok

**Comment by webbdays** [comment]
I cannot understand what is the error in lines 46,47,48 of bugbug/test_scheduling.py file.

**Comment by marco-c** [comment]
There is an unexpected empty line

**Comment by marco-c** [comment]
So you just need to remove those unneeded empty lines

File: tests/test_test_scheduling.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
 from datetime import datetime
-from typing import List
+
```
Comment (marco-c) [inline tests/test_test_scheduling.py:7]: Here's an empty line

File: tests/test_phabricator.py [hunk 2]
```diff
@@ -4,14 +4,14 @@
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
 from datetime import timedelta
-from typing import List
+
```
Comment (marco-c) [inline tests/test_phabricator.py:7]: Here's an empty line

File: scripts/past_bugs_by_unit.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 import json
 import logging
 from collections import defaultdict
-from typing import Dict, List
+
```
Comment (marco-c) [inline scripts/past_bugs_by_unit.py:10]: Here's an empty line

**Comment by marco-c** [comment]
Other than the empty lines, it seems the error is on:
```
PushResult = tuple[
    tuple[Revision],
    Revision,
    tuple[Runnable, ...],
    tuple[Runnable, ...],
    tuple[Runnable, ...],
]
```

For some reason, mypy doesn't understand the "...".
I've updated mypy in the pre-commit configuration file. Could you try to rebase your patch on top of master?

**Comment by webbdays** [comment]
Yes, that is the problem

**Comment by webbdays** [comment]
Pull from upsteam to update master and merge to working branch and push to fork. ?

**Comment by marco-c** [comment]
Yes, or you can directly rebase your branch:
```
git pull upstream master --rebase
git push origin dict --force
```

(assuming `upstream` is this repo and `origin` is your fork)

**Comment by webbdays** [comment]
Ok

File: scripts/past_bugs_by_unit.py (hunk 3 above)
Comment (webbdays) [reply scripts/past_bugs_by_unit.py:10]: I think the pre-commit removes that.?

Comment (marco-c) [reply scripts/past_bugs_by_unit.py:10]: Yes, it should

**Comment by webbdays** [comment]
its again the same problem with "..."

**Comment by marco-c** [comment]
> its again the same problem with "..."

:(
Let's keep typing.Tuple only for that one then. We'll change it later on.

**Comment by webbdays** [comment]
Ok

**Comment by webbdays** [comment]
Please look at this

https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html

**Comment by webbdays** [comment]
... with tuple is for python 3.9+

**Comment by marco-c** [comment]
We are using 3.9, so it should work. There must be a bug in mypy.
Anyway, let's keep typing.Tuple for those three specific cases for now.

**Comment by webbdays** [comment]
Ok

**Comment by marco-c** [comment]
Linting is still failing because of some unnecessary empty lines and white spaces. You can see for yourself like I described in this earlier comment: https://github.com/mozilla/bugbug/pull/2640#issuecomment-981377790.

**Comment by webbdays** [comment]
Ok

**Comment by webbdays** [comment]
Will Lint errors corrected by the pre-commit in that check(running in browser) added to the latest commit ?

**Comment by webbdays** [comment]
In pre-commit check, black check is failing.
I am using Ubuntu on Termux.
I came to know that there is problem with multiprocessing.synchronize on Termux, it says named semaphores unavailable.

**Comment by marco-c** [comment]
You can apply the fixes manually for now.
For next time, if you install pre-commit in the repo (pre-commit install), it will automatically run *before* you commit and apply some changes automatically.

**Comment by webbdays** [comment]
I have done that and pre-commit is running when I try to commit.
There black check is failing, so no commit is performed.

**Comment by marco-c** [comment]
When pre-commit fails, it either suggests you what fixes to apply, or it applies them automatically.
After you have applied your fixes, you should `git add` them and then try to commit again.

**Comment by marco-c** [comment]
There are still some linting failures

**Comment by webbdays** [comment]
Locally, all passed for pre-commit before commit.

**Comment by marco-c** [comment]
> Locally, all passed for pre-commit before commit.

It isn't possible, maybe you forgot to `git add` some of your changes before committing again?

File: bugbug/github.py [hunk 4]
```diff
@@ -67,7 +67,7 @@ def fetch_events(self, events_url: str) -> list:
 
     def fetch_issues(
         self, url: str, retrieve_events: bool, params: dict = None
-    ) -> Tuple[List[IssueDict], dict]:
+    ) -> tuple[ list[IssueDict], dict]:
```
Comment (marco-c) [inline bugbug/github.py:70]: For example here there is an unnecessary whitespace.

**Comment by webbdays** [comment]
I have patched dict to dict.patch (git diff master dict > dict.patch)
Old dict removed.
New dict is created.
Applied patch in dict.
All modified files are staged.
And commit is performed.
So, pre-commit is automatically performed.
Import error when black is running.
Error is related to import from _multiprocessing .

**Comment by marco-c** [comment]
If you have a problem while using pre-commit, you can apply the suggested fixes from https://community-tc.services.mozilla.com/tasks/SvHj7rG5RoqAj2wn4_Nddw/runs/0/logs/public/logs/live.log manually.

**Comment by webbdays** [comment]
Ok. How to do that.
Sign up in that website?

**Comment by marco-c** [comment]
No, the fixes are suggested in the textual logs directly.
For example, https://community-tc.services.mozilla.com/tasks/SvHj7rG5RoqAj2wn4_Nddw/runs/0/logs/public/logs/live.log#L350 means you have to replace `    ) -> tuple[ list[IssueDict], set]:` with `    ) -> tuple[list[IssueDict], set]:` (that is, remove the unneeded whitespace).

**Comment by webbdays** [comment]
Ok

**Review by marco-c — APPROVED** [review]
Thanks!

**Comment by webbdays** [comment]
Thanks for the merge.
I think this can be marked as my first open source contribution.
My contribution in this pull request is very basic replace task.
But in this process I have learned lot of things under your @marco-c guidance.
I have improved my knowledge of git and some Linux terminal commands.
Thanks for your guidance.

**Comment by marco-c** [comment]
Thank you for your contribution and the willpower to finish it!

**Pull request merged**